trie/zk_trie_node_test.go

```diff
@@ -0,0 +1,289 @@
+package trie
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	zkt "github.com/scroll-tech/zktrie/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewEmptyNode(t *testing.T) {
+	node := NewEmptyNode()
+	assert.Equal(t, NodeTypeEmpty, node.Type)
+
+	hash, err := node.NodeHash()
+	assert.NoError(t, err)
+	assert.Equal(t, &zkt.HashZero, hash)
+}
+
+func TestNewLeafNode(t *testing.T) {
+	k := zkt.NewHashFromBytes([]byte{1, 2, 3, 4, 5})
+	vp := make([]zkt.Byte32, 1)
+	node := NewLeafNode(k, 1, vp)
+	assert.Equal(t, NodeTypeLeaf, node.Type)
+	assert.Equal(t, uint32(1), node.CompressedFlags)
+	assert.Equal(t, vp, node.ValuePreimage)
+
+	hash, err := node.NodeHash()
+	assert.NoError(t, err)
+	assert.Equal(t, "0000000000000000000000000000000000000000000000000000000118eef34a", hash.Hex())
+}
+
+func TestNewParentNode(t *testing.T) {
+	l := zkt.NewHashFromBytes([]byte{1, 2, 3, 4, 5})
+	r := zkt.NewHashFromBytes([]byte{5, 4, 3, 2, 1})
+	node := NewParentNode(l, r)
+	assert.Equal(t, NodeTypeParent, node.Type)
+	assert.Equal(t, l, node.ChildL)
+	assert.Equal(t, r, node.ChildR)
+
+	hash, err := node.NodeHash()
+	assert.NoError(t, err)
+	assert.Equal(t, "00000000000000000000000000000000000000000000000000000000b52462b9", hash.Hex())
+}
+
+func TestNewParentNodeWithEmptyChild(t *testing.T) {
+	l := zkt.NewHashFromBytes([]byte{1, 2, 3, 4, 5})
+	r := NewEmptyNode().nodeHash
+	node := NewParentNode(l, r)
+
+	assert.Equal(t, NodeTypeParent, node.Type)
+	assert.Equal(t, l, node.ChildL)
+	assert.Equal(t, r, node.ChildR)
+
+	// Will crush: node.ChildR is nil
```

Comment:
@noel2004 `node.NodeHash()` will encounter a nil pointer error, how can I resolve this?

Reply:
Use `r, _ := NewEmptyNode().NodeHash()`, i.e. the `NodeHash` method rather than `nodeHash` field